test/blackbox-tests/test-cases/read-only-symlink-target.t/run.t

```diff
@@ -0,0 +1,22 @@
+Nix can leave a symlink to a store path in the tree, often called 'result'.
+'dune fmt' crashes because of that.
+
+  $ RESULT=`mktemp -d`
+  $ echo "let x  = 2" > "$RESULT/foo.ml"
+  $ chmod -R a-w "$RESULT"
+  $ ln -s "$RESULT" result
+
+This command should succeed:
+
+  $ dune fmt
+  File "result/foo.ml", line 1, characters 0-0:
+  Error: Files _build/default/result/foo.ml and
+  _build/default/result/.formatted/foo.ml differ.
+  Promoting _build/default/result/.formatted/foo.ml to result/foo.ml.
+  Error: failed to promote result/foo.ml
+  Permission denied
```

Comment:
Thanks for the fix! With this output, it's clearer that removing `result` would fix the issue.
But couldn't Dune avoid this issue ? Formatting files that are outside the workspace is surprising but could be part of someone's workflow. However, formatting files that are read-only feels like something Dune could avoid without bothering anyone.

Reply:
> However, formatting files that are read-only feels like something Dune could avoid without bothering anyone.

That's an interesting suggestion. It does seem like an improvement for the current behavior. There's no need to setup rules if we know they're good to fail. Do you want to try implementing this?

Reply:
I've implemented that in https://github.com/ocaml/dune/pull/11490 but I have the feeling that is not the right solution. `result` links might interfere with other things than formatting and I think they should not be read by Dune.